### PR TITLE
Course User Badge Views

### PR DESCRIPTION
**Pull request by weiqingtoh**
Dependent on #628

Mainly want to check with @allenwq on the helper methods and views for the course user badge. Are the files in the right place? 

If okay, I'll do one proper one with specs (this one has no specs yet) then do a proper PR!

### CODE REVIEW

File: app/helpers/course/controller_helper.rb [hunk 1]
```diff
@@ -46,4 +46,30 @@ def link_to_user(user, options = {}, &block)
       super(user, options, &block)
     end
   end
+
+  # Display the course_user_badge given the course and course_user
+  #
+  # @return [String] HTML string to render the course_user_badge
+  def display_course_user_badge
+    exp = current_course_user.experience_points
```
Comment (allenwq) [inline app/helpers/course/controller_helper.rb]: current_course_user might be nil ? let's say an admin visits the page.

File: app/controllers/course/controller.rb [hunk 2]
```diff
@@ -38,6 +38,7 @@ def current_component_host
       Course::ControllerComponentHost.new(current_tenant.settings(:components),
                                           current_course.settings(:components), self)
   end
+  helper_method :current_component_host
```
Comment (allenwq) [inline app/controllers/course/controller.rb]: You can get the component host by `controller.current_component_host`, think you don't need to make this a helper.

**Comment by allenwq** [comment]
@weiqingtoh Yes the files are in the right place :+1:

File: app/helpers/course/controller_helper.rb (hunk 1 above)
Comment (weiqingtoh) [inline app/helpers/course/controller_helper.rb]: @allenwq I added the logic at the view `app/views/layouts/course.html.slim`

```ruby
- unless current_course_user.staff?
  = display_course_user_badge
```

From your concern, I am thinking that we can pass `current_course_user` as a parameter into this method, so it is more explict. What do you think?

Comment (allenwq) [inline app/helpers/course/controller_helper.rb]: But `nil.staff?` will still give exceptions ?

Yep, I think that make sense. and it's consistent with our user helpers: `display_user(user)`

**Comment by weiqingtoh** [comment]
Updated - dependent on #724, to see if we can find a neater way to express the raw SQL.

File: app/views/layouts/course.html.slim [hunk 3]
```diff
@@ -3,6 +3,13 @@
   = render_breadcrumbs
   div.row
     div.col-lg-2.col-md-3.col-sm-4
+      div.container-fluid
+        div.row
+          div.col-xs-12.user
+            = display_user_image(current_user)
+            = link_to_user(current_user)
+        - unless current_course_user.nil? or current_course_user.staff?
```
Comment (lowjoel) [inline app/views/layouts/course.html.slim]: `- if current_course_user.present? && !current_course_user.staff?`

File: app/assets/stylesheets/course/layout.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#course-user-badge {
```
Comment (lowjoel) [inline app/assets/stylesheets/course/layout.scss]: Wrap this within the controller class.

**Pull request merged**